RecoTracker/MkFitCore/interface/Track.h

```diff
@@ -0,0 +1,683 @@
+
+    void setDuplicateValue(bool d) { status_.duplicate = d; }
+    bool getDuplicateValue() const { return status_.duplicate; }
+    enum class ProdType { NotSet = 0, Signal = 1, InTimePU = 2, OutOfTimePU = 3 };
+    ProdType prodType() const { return ProdType(status_.prod_type); }
+    void setProdType(ProdType ptyp) { status_.prod_type = static_cast<unsigned int>(ptyp); }
+
+    int getNSeedHits() const { return status_.n_seed_hits; }
+    void setNSeedHits(int n) { status_.n_seed_hits = n; }
+    int getEtaRegion() const { return status_.eta_region; }
+    void setEtaRegion(int r) { status_.eta_region = r; }
+
+    // Those are defined in Track, TrackCand has separate member. To be consolidated but
+    // it's a binary format change.
+    // int  nOverlapHits()  const  { return status_.n_overlaps; }
+    // void setNOverlapHits(int n) { status_.n_overlaps = n; }
+
+    /// track algorithm; partial copy from TrackBase.h
+    enum class TrackAlgorithm {
+      undefAlgorithm = 0,
+      ctf = 1,
+      duplicateMerge = 2,
+      cosmics = 3,
+      initialStep = 4,
+      lowPtTripletStep = 5,
+      pixelPairStep = 6,
+      detachedTripletStep = 7,
+      mixedTripletStep = 8,
+      pixelLessStep = 9,
+      tobTecStep = 10,
+      jetCoreRegionalStep = 11,
+      conversionStep = 12,
+      muonSeededStepInOut = 13,
+      muonSeededStepOutIn = 14,
+      outInEcalSeededConv = 15,
+      inOutEcalSeededConv = 16,
+      nuclInter = 17,
+      standAloneMuon = 18,
+      globalMuon = 19,
+      cosmicStandAloneMuon = 20,
+      cosmicGlobalMuon = 21,
+      // Phase1
+      highPtTripletStep = 22,
+      lowPtQuadStep = 23,
+      detachedQuadStep = 24,
+      reservedForUpgrades1 = 25,
+      reservedForUpgrades2 = 26,
+      bTagGhostTracks = 27,
+      beamhalo = 28,
+      gsf = 29,
+      // HLT algo name
+      hltPixel = 30,
+      // steps used by PF
+      hltIter0 = 31,
+      hltIter1 = 32,
+      hltIter2 = 33,
+      hltIter3 = 34,
+      hltIter4 = 35,
+      // steps used by all other objects @HLT
+      hltIterX = 36,
+      // steps used by HI muon regional iterative tracking
+      hiRegitMuInitialStep = 37,
+      hiRegitMuLowPtTripletStep = 38,
+      hiRegitMuPixelPairStep = 39,
+      hiRegitMuDetachedTripletStep = 40,
+      hiRegitMuMixedTripletStep = 41,
+      hiRegitMuPixelLessStep = 42,
+      hiRegitMuTobTecStep = 43,
+      hiRegitMuMuonSeededStepInOut = 44,
+      hiRegitMuMuonSeededStepOutIn = 45,
+      algoSize = 46
+    };
+
+    int algoint() const { return status_.algorithm; }
+    TrackAlgorithm algorithm() const { return TrackAlgorithm(status_.algorithm); }
+    void setAlgorithm(TrackAlgorithm algo) { status_.algorithm = static_cast<unsigned int>(algo); }
+    void setAlgoint(int algo) { status_.algorithm = algo; }
+    // To be used later
+    // bool isStopped() const { return status_.stopped; }
+    // void setStopped()      { status_.stopped = true; }
```

Comment:
possibly (a similar future use note is in place also for the `Status::stopped` member).
Since that one is a bitfield, to preserve bw compatibility with standalone bin files, we prefer to keep the `Status::stopped` member.

Considering the situation with nOverlaps (which has a different implementation for the derived types `Track` and `TrackCand`) as a precedent, there is a slight preference not to uncomment this here until there is a use case.